### ISSUE TITLE
[Android] native crash 

### ISSUE THREAD

**Issue by cmxyzx** [mapbox/mapbox-gl-native#10954]
**Platform:Android**
**Mapbox SDK version:5.3.1**

### Steps to trigger behavior

 1.build mapbox sdk on branch `release-agua`  commit `2478942`

```
# DefaultFileSour(10525)
SIGABRT

#00 pc 0003b268 /system/lib/libc.so (tgkill+12) [armeabi-v7a]
#01 pc 0001781d /system/lib/libc.so (pthread_kill+52) [armeabi-v7a]
#02 pc 00018427 /system/lib/libc.so (raise+10) [armeabi-v7a]
#03 pc 00014c77 /system/lib/libc.so (__libc_android_abort+34) [armeabi-v7a]
#04 pc 00013058 /system/lib/libc.so (abort+4) [armeabi-v7a]
#05 pc 00015e81 /system/lib/libc.so (__libc_fatal+16) [armeabi-v7a]
#06 pc 00014d01 /system/lib/libc.so (__assert2+20) [armeabi-v7a]
#07 pc 003443ef libmapbox-gl.so abort_message (/Volumes/Android/buildbot/src/android/ndk-release-r16/external/libcxx/../../external/libcxxabi/src/abort_message.cpp:73) [armeabi-v7a]
#08 pc 003444d3 libmapbox-gl.so demangling_terminate_handler (/Volumes/Android/buildbot/src/android/ndk-release-r16/external/libcxx/../../external/libcxxabi/src/cxa_default_handlers.cpp:68) [armeabi-v7a]
#09 pc 00342899 libmapbox-gl.so __terminate (/Volumes/Android/buildbot/src/android/ndk-release-r16/external/libcxx/../../external/libcxxabi/src/cxa_handlers.cpp:65) [armeabi-v7a]
#10 pc 0034231d libmapbox-gl.so __cxa_throw (/Volumes/Android/buildbot/src/android/ndk-release-r16/external/libcxx/../../external/libcxxabi/src/cxa_exception.cpp:227 [Inline: failed_throw]) [armeabi-v7a]
#11 pc 00014361 /data/app/com.moji.mjweather-1/lib/arm/libmapbox-gl.so [armeabi-v7a]
#12 pc 000ebacd libmapbox-gl.so HTTPRequest (../../../../../../../mason_packages/headers/jni.hpp/3.0.0/include/jni/class.hpp:53 [Inline: NewObject<long long, jni::jstring *, jni::jstring *, jni::jstring *>]) [armeabi-v7a]
#13 pc 000eca39 libmapbox-gl.so request (/Users/di.duan/Library/Android/sdk/ndk-bundle/sources/cxx-stl/llvm-libc++/include/memory:3026) [armeabi-v7a]
#14 pc 000dda43 libmapbox-gl.so activateRequest (/Users/di.duan/Documents/DEV/mapbox-gl-native/platform/default/online_file_source.cpp:130) [armeabi-v7a]
#15 pc 000de229 libmapbox-gl.so activateOrQueueRequest (/Users/di.duan/Documents/DEV/mapbox-gl-native/platform/default/online_file_source.cpp:109) [armeabi-v7a]
#16 pc 000de1e5 libmapbox-gl.so operator() (/Users/di.duan/Library/Android/sdk/ndk-bundle/sources/cxx-stl/llvm-libc++/include/functional:1562 [Inline: __call<(lambda at /Users/di.duan/Documents/DEV/mapbox-gl-native/platform/default/online_file_source.cpp:325:44) &>]) [armeabi-v7a]
#17 pc 000dbc5f libmapbox-gl.so operator() (/Users/di.duan/Library/Android/sdk/ndk-bundle/sources/cxx-stl/llvm-libc++/include/functional:1924) [armeabi-v7a]
#18 pc 0026d361 libmapbox-gl.so runTask (/Users/di.duan/Documents/DEV/mapbox-gl-native/platform/android/src/timer.cpp:50) [armeabi-v7a]
#19 pc 0026b235 libmapbox-gl.so processRunnables (/Users/di.duan/Documents/DEV/mapbox-gl-native/platform/android/src/run_loop.cpp:184) [armeabi-v7a]
#20 pc 0026b63f libmapbox-gl.so run (/Users/di.duan/Documents/DEV/mapbox-gl-native/platform/android/src/run_loop.cpp:236) [armeabi-v7a]
#21 pc 000d8a87 libmapbox-gl.so operator() (../../../../../../../include/mbgl/util/thread.hpp:56) [armeabi-v7a]
#22 pc 000d89d9 libmapbox-gl.so __thread_proxy<std::__ndk1::tuple<std::__ndk1::unique_ptr<std::__ndk1::__thread_struct, std::__ndk1::default_delete<std::__ndk1::__thread_struct> >, (lambda at ../../../../../../../include/mbgl/util/thread.hpp:46:30)> > (/Users/di.duan/Library/Android/sdk/ndk-bundle/sources/cxx-stl/llvm-libc++/include/thread:201 [Inline: __libcpp_tls_set]) [armeabi-v7a]
#23 pc 00016fab /system/lib/libc.so (__pthread_start(void*)+30) [armeabi-v7a]
#24 pc 00014fcf /system/lib/libc.so (__start_thread+6) [armeabi-v7a]
java:
java pending exception:
[Native crash above happened with a java pending exception as following]
java.lang.NullPointerException: Attempt to invoke interface method 'okhttp3.y okhttp3.e.a()' on a null object reference
com.mapbox.mapboxsdk.http.HTTPRequest.void <init>(long,java.lang.String,java.lang.String,java.lang.String)(HTTPRequest.java:106)
```

**Comment by tobrun** [comment]
Thank you for the report, the following lines seem to be obfuscated by proguard:

```
java.lang.NullPointerException: Attempt to invoke interface method 'okhttp3.y okhttp3.e.a()' on a null object reference
com.mapbox.mapboxsdk.http.HTTPRequest.void <init>(long,java.lang.String,java.lang.String,java.lang.String)(HTTPRequest.java:106)
```

Could you please provide the mapping for those classes, thank you

**Comment by cmxyzx** [comment]
@tobrun proguard mapping as below
```
okhttp3.Request -> okhttp3.y:

okhttp3.Call -> okhttp3.e:
    okhttp3.Request request() -> a
    okhttp3.Response execute() -> b
    void enqueue(okhttp3.Callback) -> a
    void cancel() -> c
```

**Comment by tobrun** [comment]
This is fixed with https://github.com/mapbox/mapbox-gl-native/pull/10987 and is available in the 5.3.2 release.
